Pocket/pocket-ios
# Native Collections - Save/archive stories

## Summary
* This PR adds the save/archive features from the "save" button on collection stories

## References 
* branch name

## Implementation Details
* Update `CollectionViewModel`, capture snapshot incremental differences and reload/reconfigure them
* delete `StoryModel` in favor of directly using `CollectionStory`

## Test Steps
* go to Home or Saves and open a collection
* Verify that you can save/archive stories
* Go to Saves or Home/Recent Saves and verify that the saved stories appear in there

## PR Checklist:

## Screenshots


https://github.com/Pocket/pocket-ios/assets/34376330/e719e218-2c4f-45f3-8ecb-0cd0fb979558
